lib/logging/addon/components/logging/new-edit/component.js

```diff
@@ -1,82 +1,269 @@
+  testing:                 false,
+  testOk:                  true,
+  tested:                  false,
+
+  pageScope:               reads('scope.currentPageScope'),
+  cluster:                 reads('scope.currentCluster'),
+  project:                 reads('scope.currentProject'),
+  clusterTargetType:       reads('clusterLogging.targetType'),
+  originalModelTargetType: reads('originalModel.targetType'),
+
+  init() {
+    this._super(...arguments)
+    if (get(this, 'originalModelTargetType') === 'customTarget') {
+      const sslTargetType = get(this, 'originalModel.sslTargetType')
+
+      setProperties(this, {
+        pasteOrUpload: true,
+        customContent: get(this, 'model.customTargetConfig.content'),
+      })
+
+      set(this, 'model.customTarget.config', {
+        clientKey:   get(this, `model.${ sslTargetType }Config.clientKey`),
+        clientCert:  get(this, `model.${ sslTargetType }Config.clientCert`),
+        certificate: get(this, `model.${ sslTargetType }Config.certificate`),
+      })
+    } else {
+      set(this, 'customContent', get(this, 'model.customTarget.config.content'))
+    }
+  },
 
   actions: {
-    save(cb) {
-      let targetType = get(this, 'targetType');
-      const pageScope = this.get('pageScope');
-      const model = get(this, 'model');
-      const intl = get(this, 'intl')
+    test() {
+      if (get(this, 'testing') || get(this, 'tested')) {
+        return resolve();
+      }
+      const clone = get(this, 'model').clone()
+
+      const ok = this.willSave();
 
-      // set projectId or clusterId
-      if (pageScope === 'project') {
-        set(model, 'projectId', get(this, 'project.id'));
+      if (!ok) {
+        return resolve();
       }
-      if (pageScope === 'cluster') {
-        set(model, 'clusterId', get(this, 'cluster.id'));
+
+      const data = get(this, 'model');
+      const gs = get(this, 'globalStore');
+      const pageScope = this.get('pageScope');
+
+      set(this, 'testing', true);
+
+      let url = `${ pageScope }loggings?action=test`
+
+      if (get(this, 'targetType') === 'customTarget') {
+        url = `${ pageScope }loggings?action=dryRun`
       }
 
-      model.setProperties({
-        elasticsearchConfig:    null,
-        splunkConfig:           null,
-        kafkaConfig:            null,
-        syslogConfig:           null,
-        fluentForwarderConfig:  null,
+      return gs.rawRequest({
+        url,
+        method: 'POST',
+        data,
+      }).then(() => {
+        setProperties(this, {
+          testOk:  true,
+          errors:  null,
+        });
+      }).catch((xhr) => {
+        setProperties(this, {
+          testOk:  false,
+          errors:  [xhr.body.message || xhr.body.code],
+        });
+      }).finally(() => {
+        setProperties(this, {
+          tested:  true,
+          testing: false,
+          model:   clone,
+        })
+        setTimeout(() => {
+          set(this, 'tested', false);
+        }, 3000);
```

Comment:
why are we setting a time out here? If this isn't firing in the correct work queue perhaps a `next` would work better?